KOGITO-418 : Missing structureref type for service task in VSCode BPMN Editor

inodeman opened the pull request:
Hi @hasys @romartin this is the PR for the change, I moved several folders, wanted to trigger builds to see if any errors. I am waiting for feedback from Maciej, Kris to see if my implementation of using the first Input/Output parameters data type as InMessageType/OutMessageType respectevely

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins execute full downstream build

hasys commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml:
```diff
@@ -39,7 +39,7 @@
     <module>kie-wb-common-stunner-bpmn-project-client</module>
     <module>kie-wb-common-stunner-bpmn-project-backend</module>
-    <module>kie-wb-common-stunner-bpmn-kogito-runtime</module>
+    <module>kie-wb-common-bpmn-kogito-runtime</module>
```
Hi @inodeman, why did you rename this module? now it doesn't looks consistent with other modules.

romartin replied:
Hey @hasys 

Well I proposed renaming only the folder name to @inodeman , although not being consistent yet (it was not before neither) with others, because at least this way it's automatically and properly integrated with IntelliJ projects. and the maven module id reamins the same... 

Buuut TBH now just realized this `kie-wb-common-stunner-bpmn-kogito-runtime` module is not really being used outside Stunner, because it's a webapp, it's not packaged as a JAR to be added as dependency (at least yet). Sooo probably it makes sense, and has no impact on others, it makes sense @inodeman to make the folder and/or maven module id consistent with others yeah... is should be quite straigh forward right? You think are we still on time?

inodeman replied:
Hi @romartin, if that is the case, then it is easier just to change the module name, changed in next commit

hasys commented:
Jenkins execute full downstream build

hasys commented:
Jenkins retest please

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

romartin commented:
Hey @krisv 
Although having this fix we're not sure that will work on the kogito runtime. Is it hard to test locally? I remember @DuncanDoyle pointing out some personal repository that was running into that issue, could we use it to test the fix end-to-end? Does that makes sense?

Thx guys!

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

hasys commented:
Hi @inodeman, @romartin, @krisv,

not sure is it ready to test or there are some unresolved questions?

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-bpmn-kogito-runtime/src/test/java/org.kie.workbench.common.stunner.kogito.client/editor/BPMNDiagramEditorTest.java:
```diff
@@ -0,0 +1,156 @@
+import static org.jgroups.util.Util.assertEquals;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class BPMNDiagramEditorTest {
```
I think this class is created in a wrong folder - instead of using the package name as for the forlder's name it should be using nested subfolders...

inodeman replied:
Hi @romartin moved to right folder in next commit

inodeman commented:
Jenkins execute full downstream build

inodeman commented:
Jenkins retest please

romartin commented:
Hey @hasys - can you plz do this revew ASAP? This issue is relevant for next kogito release, and we should have it previously merged during this week... thanks!

inodeman commented:
Hi @romartin, for the renamind, as you said to be consistent I changed the name of the module in the pom, with this fix, it works with intelliJ, otherwise if I changed the directory name I needed to change a whole bunch of files, let me know if ok

hasys commented:
Hi @DuncanDoyle, as @romartin mentioned [here](https://github.com/kiegroup/kie-wb-common/pull/3053#issuecomment-561817273) it will be very helpful if you have some repo to test this changes as you provided for [KOGITO-630](https://issues.redhat.com/browse/KOGITO-630). Thank you!

romartin commented:
hey @hasys  - BTW how are you testing these kogito sutff, by building it locally too? Maybe we should ask for the bpmn kogito webapp being published on jenkins, as well as we do with the business-central... does that makes sense? CC @mareknovotny @mbiarnes @krisv

hasys commented:
Hi @romartin right I compiled it locally, as described here: https://github.com/kiegroup/kie-wb-common/pull/3068

If I recall correctly @ederign mentioned that it is a plan to introduce kogito editors building as part of Full Downstream Build (or similar mechanism) in the very near future.

hasys commented:
Hi @romartin, @inodeman, I investigated a bit missing **extensionElements**[1], there are results:
### Test Procedure 1
1. import file from KOGITO-418 to new and legacy designers (in Business Central)
2. export file back and check Sequence flows

#### Actual Result
* `extensionElements` lost in both cases after export from legacy and new designers.

### Test Procedure 2
1. Create _Start->General Service Task->End_ process in both legacy and new designers.

#### Actual Result
* In the process file from the new designer Sequence Flows  have `extensionElements` element
* In the process file from the legacy designer Sequence Flow **doesn't** have `extensionElements`

### Test Procedure 3
1. Load created process in NEW designer from the **Test Procedure 2** to NEW designer and export it again

#### Actual Result
* In the process file from the new designer Sequence Flows  have `extensionElements` element when re-exported from NEW designer 

### Test Procedure 4
1. Load created process in NEW designer from the **Test Procedure 2** to LEGACY designer and export it again

#### Actual Result
* In the process file from the new designer Sequence Flows **doesn't** have `extensionElements` element when re-exported from LEGACY designer 


I tried Procedure 3 and 4 for process modelled in legacy designer but, I guess, since `extensionElements` missed in original process it's missing in resulted processes as well.

So, to me it looks like it is how new designer store data about how sequence flow's connected to the node, and it is not critical if it will be lost, but it's quite unclear why this data is lost when we import file from the provided file.

[1]
<details>
  <summary>Sequence flow with extension elements</summary>
  <pre>bpmn2:sequenceFlow id=&#x22;_C15A51AF-DECF-454E-8DAA-52C1B4D5E25C&#x22; sourceRef=&#x22;_25ADBBE9-1825-4910-99C5-1409422D46AB&#x22; targetRef=&#x22;_3636D882-5DC6-4BEC-BC37-A00F3ED38FF6&#x22;&#x3E;
      &#x3C;bpmn2:extensionElements&#x3E;
        &#x3C;drools:metaData name=&#x22;isAutoConnection.source&#x22;&#x3E;
          &#x3C;drools:metaValue&#x3E;
            &#x3C;![CDATA[true]]&#x3E;
          &#x3C;/drools:metaValue&#x3E;
        &#x3C;/drools:metaData&#x3E;
        &#x3C;drools:metaData name=&#x22;isAutoConnection.target&#x22;&#x3E;
          &#x3C;drools:metaValue&#x3E;
            &#x3C;![CDATA[true]]&#x3E;
          &#x3C;/drools:metaValue&#x3E;
        &#x3C;/drools:metaData&#x3E;
      &#x3C;/bpmn2:extensionElements&#x3E;
    &#x3C;/bpmn2:sequenceFlow&#x3E;
    &#x3C;bpmn2:sequenceFlow id=&#x22;_4C4024AC-D6CA-4B58-BCEF-D667DA948201&#x22; sourceRef=&#x22;_578F1284-FFB5-4A5A-B628-A27238DD2B5B&#x22; targetRef=&#x22;_25ADBBE9-1825-4910-99C5-1409422D46AB&#x22;&#x3E;
      &#x3C;bpmn2:extensionElements&#x3E;
        &#x3C;drools:metaData name=&#x22;isAutoConnection.source&#x22;&#x3E;
          &#x3C;drools:metaValue&#x3E;
            &#x3C;![CDATA[true]]&#x3E;
          &#x3C;/drools:metaValue&#x3E;
        &#x3C;/drools:metaData&#x3E;
        &#x3C;drools:metaData name=&#x22;isAutoConnection.target&#x22;&#x3E;
          &#x3C;drools:metaValue&#x3E;
            &#x3C;![CDATA[true]]&#x3E;
          &#x3C;/drools:metaValue&#x3E;
        &#x3C;/drools:metaData&#x3E;
      &#x3C;/bpmn2:extensionElements&#x3E;
    &#x3C;/bpmn2:sequenceFlow&#x3E;</pre>
</details>
